drivers/ethernet/eth_numaker.c

```diff
@@ -0,0 +1,807 @@
+	}
+
+	/* length of payload should be <= 1514 */
+	if (*len > (NU_ETH_MAX_FLEN - 4)) {
+		LOG_DBG("unexpected long packet length=%d, buf=0x%x", *len, (uint32_t)*buf);
+		*len = 0; /* Skip this unexpected long packet */
+	}
+	LOG_DBG("---End---");
+	return 0;
+}
+
+static void m_numaker_gmacdev_rx_next(synopGMACdevice *gmacdev)
+{
+	LOG_DBG("RX Next");
+	/* Already did in synop_handle_received_data
+	 * No-op at this stage
+	 * DmaDesc * rxdesc = (gmacdev->RxBusyDesc - 1);
+	 * rxdesc->status = DescOwnByDma;
+	 */
+}
+
+static void m_numaker_gmacdev_trigger_rx(synopGMACdevice *gmacdev)
+{
+	LOG_DBG("---Start---");
+
+	/* Enable the interrupt */
+	synopGMAC_enable_interrupt(gmacdev, DmaIntEnable);
+
+	/* Trigger RX DMA */
+	synopGMAC_enable_dma_rx(gmacdev);
+	synopGMAC_resume_dma_rx(gmacdev);
+	LOG_DBG("resume RX DMA");
+	LOG_DBG("---End---");
+}
+
+static void m_numaker_gmacdev_packet_rx(const struct device *dev)
+{
+	struct eth_numaker_data *data = dev->data;
+	synopGMACdevice *gmacdev = data->gmacdev;
+	uint8_t *buffer;
+	uint16_t len = 0;
+	struct net_pkt *pkt;
+	k_spinlock_key_t key;
+
+	/* Get exclusive access, use spin-lock instead of mutex in ISR */
+	key = k_spin_lock(&data->rx_frame_buf_lock);
+
+	/* Two approach: 1. recv all RX packets in one time.
+	 *               2. recv one RX and set pending interrupt for rx-next.
+	 */
+	while (1) {
+		/* get received frame */
+		if (m_numaker_gmacdev_get_rx_buf(gmacdev, &len, &buffer) != 0) {
+			break;
+		}
+
+		if (len == 0) {
+			break;
+		}
+		/* Allocate a memory buffer chain from buffer pool
+		 * Using root iface. It will be updated in net_recv_data()
+		 */
+		pkt = net_pkt_rx_alloc_with_buffer(data->iface, len, AF_UNSPEC, 0, K_NO_WAIT);
+		if (!pkt) {
+			LOG_ERR("pkt alloc frame-len=%d failed", len);
+			goto next;
+		}
+
+		LOG_DBG("length=%d, pkt=0x%x", len, (uint32_t)pkt);
+		/* deliver RX packet to upper layer, pack as one net_pkt */
+		if (net_pkt_write(pkt, buffer, len)) {
+			LOG_ERR("Unable to write RX frame into the pkt");
+			net_pkt_unref(pkt);
+			goto error;
+		}
+		if (pkt != NULL) {
+			int res = net_recv_data(data->iface, pkt);
+
+			if (res < 0) {
+				LOG_ERR("net_recv_data: %d", res);
+				net_pkt_unref(pkt);
+				goto error;
+			}
+		}
+next:
+		m_numaker_gmacdev_rx_next(gmacdev);
+	}
+	m_numaker_gmacdev_trigger_rx(gmacdev);
+
+error:
+	k_spin_unlock(&data->rx_frame_buf_lock, key);
+}
+
+static uint8_t *m_numaker_gmacdev_get_tx_buf(synopGMACdevice *gmacdev)
+{
+
+	DmaDesc *txdesc = gmacdev->TxNextDesc;
+
+	if (!synopGMAC_is_desc_empty(txdesc)) {
+		return NULL;
+	}
+
+	if (synopGMAC_is_desc_owned_by_dma(txdesc)) {
+		return NULL;
+	} else {
+		return (uint8_t *)(txdesc->buffer1);
+	}
+}
+
+static void m_numaker_gmacdev_trigger_tx(synopGMACdevice *gmacdev, uint16_t length)
+{
+	DmaDesc *txdesc = gmacdev->TxNextDesc;
+	uint32_t txnext = gmacdev->TxNext;
+#if defined(NU_USING_HW_CHECKSUM)
+	uint32_t offload_needed = 1;
+#else
+	uint32_t offload_needed = 0;
+#endif
```

Comment:
I think you could do:

`bool offload_needed = IS_ENABLED(NU_USING_HW_CHECKSUM);`